deploy/kubernetes/elastic-agent-managed/elastic-agent-managed-daemonset.yaml

```diff
@@ -0,0 +1,80 @@
+apiVersion: apps/v1
```

Comment:
1. This might be my ignorance, but why do we keep them in two directories (`deploy/kubernetes/elastic-agent-managed` vs `deploy/kubernetes`) ?
I'm asking as most likely I'm interested in replacing the https://github.com/elastic/elastic-package/blob/master/internal/install/static_kubernetes_elastic_agent_yml.go with something more sophisticated and as small as possible.

2. Is it only an example or is it tested by CI somehow?

Reply:
1. This is a good question. In `deploy/kubernetes/elastic-agent-managed` the manifests are splited by wind while in the `deploy/kubernetes` it is just one manifest for all. I followed the pattern of the rest. Maybe this gives users more flexibility on how to deploy the manifests and which to deploy. @ChrsMark do you if there is another reason?

2. I have tested it locally with the snapshot image version and elastic stack in elastic cloud.

Reply:
1. Under `deploy/kubernetes/elastic-agent-managed` the manifests are split per kind as @MichaelKatsoulis explained. Then `deploy/kubernetes/elastic-agent-managed.yml` is the final complete manifest based on them if you run `make all` (see Makefile in same directory). To be honest I don't really know the reason for this but we keep doing like this so far.
2. CI only tests that these manifests are valid as far as I know. See lint stage at https://github.com/elastic/beats/blob/f4c570b06ff5bf8a4fd079da23df0cf10ef0318b/deploy/kubernetes/Jenkinsfile.yml#L15